polkadot/xcm/src/v3/mod.rs

```diff
@@ -60,13 +62,33 @@ pub const VERSION: super::Version = 3;
 /// An identifier for a query.
 pub type QueryId = u64;
 
-#[derive(Derivative, Default, Encode, Decode, TypeInfo)]
+#[derive(Derivative, Default, Encode, TypeInfo)]
 #[derivative(Clone(bound = ""), Eq(bound = ""), PartialEq(bound = ""), Debug(bound = ""))]
 #[codec(encode_bound())]
-#[codec(decode_bound())]
 #[scale_info(bounds(), skip_type_params(Call))]
 pub struct Xcm<Call>(pub Vec<Instruction<Call>>);
```

Comment:
Just use a `BoundedVec`. That return an error if the length you are trying is too much.

Reply:
A `BoundedVec` would break the API, right? I will do that for XCMv4 but for now this seemed like a solid solution